src/manifest.ts

```diff
@@ -0,0 +1,83 @@
+import { ManifestChunk as ViteManifestChunk } from 'vite'
+
+export interface ManifestChunk extends ViteManifestChunk {
+  isModule?: boolean
+  // https://developer.mozilla.org/en-US/docs/Web/HTML/Link_types/preload#what_types_of_content_can_be_preloaded
+  asType?: 'audio' | 'document' | 'embed' | 'fetch' | 'font' | 'image' | 'object' | 'script' | 'style' | 'track' | 'worker' | 'video'
```

Comment:
It's for both prefetch + preload...  And it's not called type because that would be confusing in a situation like:

https://github.com/nuxt-contrib/vue-bundle-renderer/blob/60b55c517a30ffd38c5f99bc7b31e83a7d85851a/src/runtime.ts#L217-L218

Other ideas welcome! (Obviously `as` is an option but potentially cryptic.)

Reply:
Yes `as` is more ctyptic even. `asType` is also cryptic for me when exposing to public...

One option could think is `linkType`. Both prefetch/preload generate link and [whatwg refering to as "link type"](https://html.spec.whatwg.org/multipage/semantics.html#attr-link-as)

Reply:
Also checking MDN docs, i don't see `as` being required for prefetching (https://developer.mozilla.org/en-US/docs/Glossary/Prefetch). More precise `preloadType` still makes sense to me honestly if usage is for links. We can sately have other `*Type`s with explicit purpose and default them to each other if reusability makes sense.

Reply:
It's not required but can be helpful. You can see the example in the docs you linked is using it when the link prefetch is rendered in headers.

Reply:
What about `hintType` / `resourceHintType`?

Reply:
Updated to `type` and `mimeType` - sound okay?

Reply:
I like `mimeType`! What I'm afraid of is that we are binding very generic `type` in manifest spec to the specific use for resource hints (preload and also prefetch). We also didn't use `type: module` but

Reply:
Understood. I think `hintType` isn't quite right. Possible values: `'audio' | 'document' | 'embed' | 'fetch' | 'font' | 'image' | 'object' | 'script' | 'style' | 'track' | 'worker' | 'video'`. See https://developer.mozilla.org/en-US/docs/Web/HTML/Element/link#attr-as - description is that this specifies the 'type of content being loaded'. Maybe `contentType` is better?

Reply:
To be exact it is `linkContentType ([Content-Type](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Type) also refers to mime in another context 😆). Do you think `resourceHintType` is not explainary? I insipred from W3C standard name for group of this features (https://w3c.github.io/resource-hints/). (I'm thinking of not mention `link` in name because it can also be a header for prefetch server hints)

Reply:
`resourceType`?

Reply:
I still prefer to be exact in name as usage and possible values are for resources hint generation. We set possible types also based on this from mdn docs. 


we don't need a generic resourceType at least today accepting things types such as `payload` for special usecases of chunks. but if at some point want it, have to choose another similar name and that makes manifest confusing. 

On the other side, i understand this type feels generic enough to have simpler name....